src/components/Carousel/Carousel.tsx

```diff
@@ -135,20 +143,57 @@ const Carousel: React.FC<CarouselProps> = ({
     setCurIndex(length - 1)
   }, [length])
 
+  /**
+   * Pause
+   */
+  const pause = useRef(false) // Flag to pause carousel loop
+
   /**
    * Slide to the previous carousel item
    */
   const slidePrev = useCallback(() => {
+    if (pause.current) return
     setCurIndex((curIndex: number) => getPrevIndex(length, curIndex))
   }, [length])
 
   /**
    * Slide to the next carousel item
    */
   const slideNext = useCallback(() => {
+    if (pause.current) return
     setCurIndex((curIndex: number) => getNextIndex(length, curIndex))
   }, [length])
 
+  const selectSlide = useCallback(
+    (index: number) => {
+      if (selectable) {
+        if (curIndex === index && onClickSelectedItem)
+          onClickSelectedItem(index)
+        else if (onClickItem) onClickItem(index)
+        setCurIndex(index)
+      }
+    },
+    [curIndex, selectable, onClickSelectedItem, onClickItem]
+  )
+
+  const startPause = useCallback(
```

Comment:
This is my opinion, but how about attaching the pauseHandlers(`startPause`, `releasePause`) to the carousel container instead of each individual carousel item?

```javascript
<div
   className={`react-responsive-3d-carousel__carousel__list ${spread}`}
   ref={containerRef}
   onPointerEnter={startPause}
   onPointerLeave={releasePause}
 >
```

```javascript
  const startPause = useCallback(() => {
    if (pauseOnHover) pause.current = true
  }, [pauseOnHover])

  const releasePause = useCallback(() => {
    if (pauseOnHover) pause.current = false
  }, [pauseOnHover])
```

Reply:
My question here is if you want to use the arrows to change the slide and don't work, but I tried it and it worked